specification/attestation.adoc

```diff
@@ -1,82 +1,245 @@
+data, by applying a cryptographic hash algorithm (`H~alg~`) to the current
+measurement value (`M~current~`):
+
+:stem: asciimath
+[.center]
+[stem]
+++++
+M_{extended} = H_{alg}(M_{current} || NewData)
+++++
+
+Measurements must be stored in integrity protected measurement registers,
+either in the Hardware RoT of Measurement (RoTM) or in the TSM when it acts as
+an extended RoTM.
+
+In order to verify a workload environment trustworthiness, the remote
+attestation service must be able to compare the AP-TEE TCB measurements with a
+set of reference values. All components of what constitues the workload Trusted
+Computing Base (TCB) must be measured and included in the attestation evidence.
+
+==== TCB Elements
+
+Elements considered to be in the TCB for AP-TEE workloads are both hardware
+and software components.
+
+The overall workload TCB is a composition of 3 independent TCBs:
+
+1. The Platform TCB is made of all hardware and software TCB elements for
+the AP-TEE host platform. They are not confidential compute specific and
+measurements for those elements could be shared with e.g. a measured boot
+attestation.
+2. The TSM TCB elements are the confidential compute specific but TVM agnostic
+elements for the overall TCB.
+3. The TVM TCB is composed of the TVM specific elements. 
+
+The TCB elements for each of them is summarized in the following table:
+
+.AP-TEE Workload TCB Elements
+[cols="^1, <5, <4"]
+|===
+|TCB |Hardware Elements |Software Elements
+
+ | Platform
+a| * HW RoT for boot, measurement and storage
+   * All CPU hardware logic, including MMU and caches
+   * All SoC subsystems, including memory confidentiality, integrity and replay-protection for volatile memory
+   * IOMMU and translation agents
+a| * All M-mode firmwares, except for the TSM-driver
+
+ | TSM
+ | N/A
+a| * TEE Security Manager (TSM) and its U-mode components
+   * TSM-Driver that hosts a TEEI (with TH-ABI and TG-ABI security routines). 
+
+ | TVM
+a| * Directly assigned, TEE-IO compliant devices
+ | * All TVM measured pages
+|===
+
+==== TVM Measurements
+
+At a high level, a TVM measurement is separated into a set of static
+measurements and a dynamic one.
+
+===== Static Measurements
+
+TVM static measurements are generated from the AP-TEE workload TCB elements
+involved in the TVM construction. Any TCB element that directly or indirectly
+supports a TVM generation must be measured into the TVM static measurement
+registers. Once a TVM is finalized, i.e. after the
+`sbi_tee_host_finalize_tvm()` TH-ABI is called, the TVM static measurements
+must no longer be extended.
+
+Static measurements are stored in dedicated measurement registers and an AP-TEE
+implementation must provide at least 1 and at most 16 of them. Some of the
+static measurement registers cover platform or TSM specific TCB elements while
+other are measuring TVM specific TCB pieces, and have respectively platform
+bound and TVM bound lifecycles. As such the former may be stored in the RoT
```

Comment:
Sounds bit ambiguous to me. As per my understanding, static measurements cover platform, tsm or tvm measurements. Correct ?

What exactly do you mean by platform
bound and TVM bound lifecycles ? Everything static measurement is platform bound. Isn't it ?

TVM measurements are TVM bound and should be stored in TSM confidential memory.

Reply:
Good suggestion, thanks. Let me know if the rewording looks better to you.